### PR TITLE
2891 - Exception manager RBAC

### PR DESCRIPTION
**Pull request by liamtoozer**
# Motivation and Context

We need to add fine-grained Role-Based Access Control permissions to our Exception Manager endpoints when accessing via the support tool.

# What has changed

* Add new activity types for RBAC
* Update SQL scripts

# How to test?

Run the UI with the branches for this ticket. Generate some bad messages and check all still works for the user's permission levels.

# Links

https://trello.com/c/QO1t6Cr7/2891-exception-manager-via-support-tool-should-have-seperate-view-errors-peek-and-quarantine-rbac-roles-add-to-support-groups-8

### CODE REVIEW

**Comment by liamtoozer** [comment]
[Exception manager via Support Tool should have seperate "view errors", "peek" and "quarantine" RBAC roles + add to support groups (8)](https://trello.com/c/QO1t6Cr7/2891-exception-manager-via-support-tool-should-have-seperate-view-errors-peek-and-quarantine-rbac-roles-add-to-support-groups-8)

File: ssdc-rm-common-entity-model/pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>uk.gov.ons.ssdc</groupId>
   <artifactId>ssdc-rm-common-entity-model</artifactId>
-  <version>3.2.2-SNAPSHOT</version>
+  <version>3.3.0-SNAPSHOT</version>
```
Comment (NickGrantONS) [inline ssdc-rm-common-entity-model/pom.xml:9]:
```suggestion
  <version>3.2.3-SNAPSHOT</version>
```

File: ui-permissions/RM-Support-Permissions.sql [hunk 2]
```diff
@@ -10,3 +10,6 @@ INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, sur
...
 INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('26cf0f7a-7f0f-4e5b-8be2-640745be9cdf', 'VIEW_SAMPLE_LOAD_PROGRESS', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
 INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('81395c04-416c-4e3f-9397-83a08ac8bdf2', 'LIST_SMS_TEMPLATES', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('6f655e60-be27-4092-84e0-3b64971a4dac', 'EXCEPTION_MANAGER_VIEWER', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('c486f714-3c64-4416-bf77-58b2f3d4d617', 'EXCEPTION_MANAGER_PEEK', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('e1a9f8cf-efbd-46b7-919c-37d33e25bc7d', 'EXCEPTION_MANAGER_QUARANTINE', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
```
Comment (NickGrantONS) [inline ui-permissions/RM-Support-Permissions.sql:15]: basic read-only support role probably shouldn't be allowed to quarantine messages

File: ui-permissions/RM-Support-Permissions.sql [hunk 3]
```diff
@@ -10,3 +10,6 @@ INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, sur
 INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('81e78da1-5e2a-4457-a2ee-cc12c884f2fa', 'LIST_ACTION_RULES', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
 INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('26cf0f7a-7f0f-4e5b-8be2-640745be9cdf', 'VIEW_SAMPLE_LOAD_PROGRESS', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
 INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('81395c04-416c-4e3f-9397-83a08ac8bdf2', 'LIST_SMS_TEMPLATES', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('6f655e60-be27-4092-84e0-3b64971a4dac', 'EXCEPTION_MANAGER_VIEWER', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('c486f714-3c64-4416-bf77-58b2f3d4d617', 'EXCEPTION_MANAGER_PEEK', 'b19a77bd-6a02-4851-8116-9e915738b700', NULL);
```
Comment (NickGrantONS) [inline ui-permissions/RM-Support-Permissions.sql:14]: basic read-only support role probably shouldn't be allowed to peek messages

Comment (liamtoozer) [reply ui-permissions/RM-Support-Permissions.sql]: I think I might have been going on what the support peeps could do previously. Makes sense tho 👍

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**